### PR TITLE
Sub_plain_inplace

### PR DESCRIPTION
**Pull request by bmcase**
separating out sub_plain_inplace update to lib.rs from the cupcake parallel PR.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -678,6 +692,36 @@ mod fv_scalar_tests {
...
+        let sk = fv.generate_key();
+
+        let mut v = vec![0; fv.n];
+        for i in 0..fv.n {
+            v[i] = i as u8;
+        }
+
+        let mut w: Vec<u8> = vec![];
+        for i in 0..fv.n {
+            w.push((fv.n + i) as u8);
```
Comment (haochenuw) [inline src/lib.rs]: fv.n is 2048 which will overflow u8 already -- maybe use some smaller numbers?

File: src/lib.rs [hunk 2]
```diff
@@ -678,6 +692,36 @@ mod fv_scalar_tests {
...
+        }
+
+        let mut vplusw = vec![];
+        for _ in 0..fv.n {
+            vplusw.push(fv.n as u8);
```
Comment (haochenuw) [inline src/lib.rs]: should we rename this as v_minus_w

File: src/lib.rs [hunk 3]
```diff
@@ -678,6 +692,36 @@ mod fv_scalar_tests {
...
+        }
+        // encrypt v
+        let mut ct = fv.encrypt_sk(&v, &sk);
+
+        // ct_v - w.
+        fv.sub_plain_inplace(&mut ct, &w);
+
+        let pt_after_add : DefaultFVPlaintext= fv.decrypt(&ct, &sk);
```
Comment (haochenuw) [inline src/lib.rs]: pt_after_subtract?

**Review by haochenuw — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 1 above)
Comment (bmcase) [reply src/lib.rs]: same as the addition tests though

Comment (haochenuw) [reply src/lib.rs]: actually, that's right! Could you fix the addition tests as well? Really appreciate it!

**Review by haochenuw — APPROVED** [review]
LGTM! Thanks for working on this!

**Pull request merged**